### DISCUSSION TITLE
Bug? Multiple duplicate groups detected

### DISCUSSION THREAD

**Discussion by BrentEubanks** [SamDel/ChromeCast-Desktop-Audio-Streamer#115] [General]
When DAS launches, it detects the same groups multiple times.  See attached image.  All these groups are legit (and its set to show only groups) but some of them are duplicates, marked error.

"scan again" button has no effect; restarting the app recreates the issue, though it's not always the same groups being duplicated.

![image](https://user-images.githubusercontent.com/25329360/127788074-593c6bc4-c661-4566-9104-739ff4924f68.png)

**Comment by SamDel** [comment]
"Reset Settings" (on the options tab) clears the list of known devices. I hope that will solve your issue.

**Comment by BrentEubanks** [comment]
That worked, sort of.

One on computer, duplicates are gone and streaming works.

One another computer, some of the groups re-duplicate after the reset.  That computer is now unable to stream to any device.  When I try, the device/group briefly flashes green before disconnecting.
Several devices show the "check firewall" error, though it is not consistent which ones show this error.  I'm not sure why the firewall would be involved - the desktop and all CCs are on the same LAN.

**Reply by SamDel** [reply]
Rebooting the other computer and/or the router/devices will probably solve it.

**Reply by BrentEubanks** [reply]
tried rebooting everything.  Still getting duplicate devices and connection failures, on both computers - I spoke too soon about one of them being fixed.

Could these errors be the result of running DAS on more than one computer in the same LAN?

**Reply by SamDel** [reply]
No, it's not the result of multiple DAS instances in the same LAN.

The devices broadcast their existence, and the groups, in the LAN. In the broadcast-message there is a name and id. In your case it looks like the id of the group has changed, but the braodcast-message with the previous id is still received by DAS. I'm not sure why, maybe it's cached somewhere...

**Reply by BrentEubanks** [reply]
Another option: I have an Orbi mesh wireless system.  It works great - and I turned off Daisy Chain, which is known to be problematic with CCs.  Could it be part of the problem?

Any idea where the old ID could be cached?

**Reply by SamDel** [reply]
I'm not sure if it's cached, singing around your network, or ...
Have you tried to turn off all Orbi devices at the same time?

**Reply by BrentEubanks** [reply]
I seem to have finally gotten this to work.  I had to unplug all the CCs, then reboot my modem, then my router.  Then replug the CCs and for now all is good. 
I had done all this before, but piecewise.  Doing everything all at once was the key.

**Reply by SamDel** [reply]
Nice!

**Reply by FA-Bubba** [reply]
FYI, I have noted when I unplug a Chromecast Audio dongle or a Google/Nest Mini or Hub, they update their System & Cast Firmware versions,  I have not discovered any other way to get them to do the updates while they are on...

So it is possible that old Firmware might be a factor in what you have experienced...

(If I am in a lazy mood, I'll just flip the Master Circuit Breaker to Off for about a minute,

**Comment by BrentEubanks** [comment]
Also: when I change the filter (to show groups/devices/both) I get an "unhandled exception error".  Not sure if that's related to not.

**Reply by FA-Bubba** [reply]
I had a problem similar to that... with multiple instances of Groups and/or Speaker Devices, mostly displaying in the "error color'.  My configuration is:
 - Mesh network with 2 APs in a 3,900 SF residence.  Internet from Comcast Gateway, which has a different SSID, which is not used by any device -- they are all set to connect to the SSID of the Mesh.
 - 12 Stereo speaker Pairs, each pair connected via Chromecast Audio
 - 4 Google Minis
 - 1 Google hub
 - All Devices are organized into 3 Groups, plus an "ALL Speakers" Group
 - 1 PC with DAS usually casting to one Group of two speaker Pairs
 - 1 PC with DAS usually casting to two Groups that include all other Speakers, Minis & Hub
 -  DAS displays ALL Groups and ALL devices, including some that are not part of any Group, and not used very often.
At times my internet provider is unstable, and that causes all or parts of the system to drop off...

While I cannot do much about the Internet Service, I have assigned Static IP addresses to every audio device on my network, and this greatly stabilized the setup.

Every now & then devices drop off and come back, which I attribute to the Internet connection (even though the LAN Mesh is very stable, it seems that DAS requires Internet connection to work properly).

When the drop-outs become troublesome, using "Reset Settings" almost always restores everything; in the infrequent times when that fails, I reboot the PCs, the Mesh and the Router/Gateway, and that has always restored everything.

**Reply by SamDel** [reply]
Thanks @FA-Bubba !

@BrentEubanks : The "unhandled exception error" shouldn't happen. Can you post the details (it should be stored in the Windows event viewer)? It's not related to the multiple groups issue.

**Reply by BrentEubanks** [reply]
@SamDel where would I find those logs in Event Viewer?  There's a lot in there.

When I find them, shall I post them to this thread, or the new one I created for the issue?

**Reply by SamDel** [reply]
In "Windows Logs" - "Application". You can post them in this thread.

**Reply by BrentEubanks** [reply]
I looked there, and I searched for ChromeCast.Desktop.AudioStreamer.  The only hit I got is from days ago - nothing from the event earlier this morning.  Is there something else I should search for?

**Reply by SamDel** [reply]
"ChromeCast-Desktop-Audio-Streamer" should find it!

**Reply by BrentEubanks** [reply]
no hits on ChromeCast-Desktop-Audio-Streamer at all.  I looked in both Application and System.

**Reply by SamDel** [reply]
Sorry, typo, I meant "ChromeCast.Desktop.AudioStreamer"!

**Reply by BrentEubanks** [reply]
OK.  That thread gets one hit, but it's days old an unrelated.

if it happens again, I'll dump and post the full contents of the "details" dropdown.  Does that work?

**Reply by SamDel** [reply]
👍
